### PR TITLE
Fix device name validation in "modelstore" command

### PR DESCRIPTION
**Pull request by alonre24**
The fifth character was not validated.
Also, for scriptstore and the deprecated modelset and scriptset, the only validation occurs in `parseDeviceStr` (which is from within the backends, after parsing stage). This PR replaces `sscanf` with `strtoll` in `parseDeviceStr`, to have safe exit in case of invalid device id string in these commands.

### CODE REVIEW

File: src/redisai.c [hunk 1]
```diff
@@ -173,12 +173,14 @@ int RedisAI_ModelStore_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **arg
...
     AC_GetString(&ac, &devicestr, NULL, 0);
     bool valid_device = false;
+    size_t device_prefix_len = strlen("GPU:"); // can also be "CPU:"
     if (strcasecmp(devicestr, "CPU") == 0 || strcasecmp(devicestr, "GPU") == 0) {
         valid_device = true;
-    } else if ((strncasecmp(devicestr, "GPU:", 4) == 0 || strncasecmp(devicestr, "CPU:", 4) == 0) &&
-               strlen(devicestr) <= 10) {
+    } else if ((strncasecmp(devicestr, "GPU:", device_prefix_len) == 0 ||
+                strncasecmp(devicestr, "CPU:", device_prefix_len) == 0) &&
+               strlen(devicestr) < 10) {
```
Comment (DvirDukhan) [inline src/redisai.c:181]: why the change from `<=` to `<`?

File: tests/flow/tests_commands.py [hunk 2]
```diff
@@ -19,13 +19,22 @@ def test_modelstore_errors(env):
...
+
+    # Check for valid device argument (should only be "CPU:<n>" or "GPU:<n>, where <n> is a number
+    # (contains digits only, up to 5).
     check_error_message(env, con, "Invalid DEVICE",
                         'AI.MODELSTORE', 'm{1}', 'TORCH', 'bad_device', 'BLOB', model_pb)
+    check_error_message(env, con, "Invalid DEVICE",
+                        'AI.MODELSTORE', 'm{1}', 'TORCH', 'CPU::', 'BLOB', model_pb)
+    check_error_message(env, con, "Invalid DEVICE",
+                        'AI.MODELSTORE', 'm{1}', 'TORCH', 'CPU:1.8', 'BLOB', model_pb)
+    check_error_message(env, con, "Invalid DEVICE",
```
Comment (DvirDukhan) [inline tests/flow/tests_commands.py:36]: add a comment explaining why this is invalid device

**Review by DvirDukhan — COMMENTED** [review]
where was the 5th character did not validated?
anyway, looks good

File: src/redisai.c (hunk 1 above)
Comment (alonre24) [reply src/redisai.c:181]: No actual reason... I believe that 5 digits should be enough

File: tests/flow/tests_commands.py (hunk 2 above)
Comment (alonre24) [reply tests/flow/tests_commands.py:36]: There is a comment in lines 28-29 explaining it...

**Review by DvirDukhan — APPROVED** [review]
